test/db/io/baddr

```diff
@@ -17,7 +17,7 @@ RUN
 
 NAME=3: io.baddr
 FILE=bins/elf/analysis/pie
-CMDS=!rz-bin -B 0xf00000 -re ${RZ_FILE} | grep 0x
+CMDS=!!rz-bin -B 0xf00000 -re ${RZ_FILE} | grep 0x
```

Comment:
I think this should be kinda automatic or, at least, the default behaviour, without having to use `!!` all the time.
If we read the help of `|` it is `| <cmd> | <program> # Pipe output of command to program`, so by using `|` the user already express his will to pipe the output of one "command" to another program. 

Do you think it would make sense to detect the `pipe` and assume output should be redirected to rzcons?

Reply:
i disagree
you can't know if this is piped, or not

Reply:
Why can't you? In `pipe_stmt` handler you know if a program is piped or not.

Reply:
yes, but what if my cmd is like 
```
[0x00000000] `aaa | bbb` | ccc
```

Reply:
Not sure about it's better `!!<prog1> | <prog2>` or `!<prog1> | <prog2>`...

Choosing `!!<prog1> | <prog2>` means that essentially you never use `!<prog1> | <prog2>` and you have to type `!!` every time. Also, as a user you first do `!ls` and then you probably want to grep it and suddenly you have to do `!!ls` instead of just `!ls` if you want to pipe. On the other hand, this maintains the semantic of `!` and `!!`.

Choosing `!<prog1> | <prog2>` and make sure that `!<prog1>` automatically redirects stdout to rzcons whenever a pipe is used means that as a user it's easier to construct a command without really having to *know* about rzcons (which is a very internal thing after all..). However, it loses a bit the semantic of `!` vs `!!`.

WDYT @wargio  @XVilka @thestr4ng3r ?

Reply:
for me you can remove `!!` and merge them into `!` and do the autodetect for pipe/grep/etc..

Reply:
I think autodetection will provide the best user experience, indeed